src/index.js

```diff
@@ -2,18 +2,23 @@ import * as actions from './actions'
 import * as selectors from './selectors'
 import reducer from './reducer'
 
-const func = middlewares => key => path => prefix => Object.assign(
-  reducer(middlewares)(key)(prefix),
-  ...Object.keys(actions).map(k => ({ [k]: actions[k](prefix) })),
-  ...Object.keys(selectors).map(k => ({ [k]: selectors[k](path)(prefix) })),
+const func = middlewares => key => path => name => prefix => Object.assign(
+  reducer(middlewares)(key)(`${prefix}${name}`),
+  ...Object.keys(actions).map(k => ({ [k]: actions[k](`${prefix}${name}`) })),
+  ...Object.keys(selectors).map(k => ({ [k]: selectors[k](path)(name) })),
 )
 
-export default (...args) => {
-  const length = args.length
-  const [middlewares, key, path, prefix] = args
 
-  if (length < 2) return func(middlewares)
```

Comment:
:cry: really, i think of the "middlewares" param is not the default case. many time i code :
`func(/*no middleware*/)('id')('path')('name')`
it's boring ....